lgc/patch/PatchBufferOp.cpp

```diff
@@ -420,12 +420,30 @@ void PatchBufferOp::visitCallInst(CallInst &callInst) {
   } else if (callName.startswith(lgcName::LateBufferLength)) {
     Value *const pointer = getPointerOperandAsInst(callInst.getArgOperand(0));
 
+    // Compute the length of the variable sized array
     // Extract element 2 which is the NUM_RECORDS field from the buffer descriptor.
-    Value *const bufferLength = m_builder->CreateExtractElement(m_replacementMap[pointer].first, 2);
+    Value *const bufferDesc = m_replacementMap[pointer].first;
+    Value *numRecords = m_builder->CreateExtractElement(bufferDesc, 2);
+
+    // The result is clamped at 0 if the offset is greater than the buffer length and robustAccess is on
```

Comment:
Hmm, re-looking at the code, my suggested comment is wrong. It should be something like "The result is 0 for a null buffer descriptor".

Reply:
Updated the comment.